vm_eval.c

```diff
@@ -639,6 +639,8 @@ rb_check_funcall(VALUE recv, ID mid, int argc, const VALUE *argv)
 static VALUE
 rb_check_funcall_default_kw(VALUE recv, ID mid, int argc, const VALUE *argv, VALUE def, int kw_splat)
 {
+    VM_ASSERT(ruby_thread_has_gvl_p());
```

Comment:
VM_ASSERT is only if `VM_CHECK_MODE > 0` so if `-DRUBY_DEBUG`, so not in regular builds.
Sounds fine to me.